test/extensions/compression/gzip/compressor/gzip_compressor_library_test.cc

```diff
@@ -0,0 +1,107 @@
+#include "extensions/compression/gzip/compressor/config.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Gzip {
+namespace Compressor {
+
+class GzipTest : public testing::Test {
```

Comment:
Setting up tests that are friend classes of the class you are testing, and exposing private member variables is not very common in Envoy. I would err on collapsing these tests on the zlib_compressor_impl_test.cc file and test for the expectations via the public API.

Reply:
Ok, merged the tests and dropped friend class from `GzipCompressorFactory`.